matter_persistence/redis/api_cache_client.py

```diff
@@ -0,0 +1,257 @@
+        return AsyncRedisClient(connection_pool=self.__connection_pool)
+
+    async def close_connection_pool(self) -> None:
+        """
+        Closes the singleton connection pool to Redis.
+
+        Since the connection pool is a singleton, this effectively stops the application process from being able
+        to connect to Redis, so use only once, when all connections should be closed!
+        """
+        await self.__connection_pool.aclose()
+
+    async def save_raw(
+        self,
+        key: str,
+        value: Any,
+        object_class: type[BaseModel] | None = None,
+        expiration_in_seconds: int | None = None,
+    ):
+        object_name = object_class.__name__ if object_class else None
+        hash_key = CacheHelper.create_basic_hash_key(key, object_name)
+        if object_class:
+            value = value.model_dump_json()
+
+        async with self.__get_cache_client() as cache_client:
+            if expiration_in_seconds:
+                result = await cache_client.set_value(hash_key, value, ttl=expiration_in_seconds)
+            else:
+                result = await cache_client.set_value(hash_key, value)
+
+        if not result:
+            raise CacheRecordNotSavedError(
+                description=f"Unable to store value in cache. Key: {key}",
+                detail={"key": key, "hash_key": hash_key},
+            )
+
+        return result
+
+    async def find_raw(self, key: str, object_class: type[BaseModel] | None = None) -> Any:
+        object_name = object_class.__name__ if object_class else None
+        hash_key = CacheHelper.create_basic_hash_key(key, object_name)
+        async with self.__get_cache_client() as cache_client:
+            value = await cache_client.get_value(hash_key)
+        if not value:
+            raise CacheRecordNotFoundError(
+                description=f"Unable to retrieve value from cache. Key: {key}",
+                detail={"key": key, "hash_key": hash_key},
+            )
+
+        if object_class:
+            if isinstance(value, list):
+                value = [object_class.parse_raw(item) for item in value]
+            else:
+                value = object_class.model_validate_json(value)
+
+        return value
+
+    async def delete_raw(
+        self,
+        key: str,
+        object_class: type[BaseModel] | None = None,
+    ) -> Any:
+        object_name = object_class.__name__ if object_class else None
+        hash_key = CacheHelper.create_basic_hash_key(key, object_name)
+
+        async with self.__get_cache_client() as cache_client:
+            if not await cache_client.delete_key(hash_key):
+                raise CacheRecordNotFoundError(
+                    description=f"Unable to retrieve value from cache. Key: {key}",
+                    detail={"key": key, "hash_key": hash_key},
+                )
+
+    async def raw_cache_record_exists(
+        self,
+        key: str,
+        object_class: object | None = None,
```

Comment:
Incorrect typing - it should be `type[Any] | None`. Please review it in the whole module.